samcli/lib/utils/path_utils.py

```diff
@@ -19,3 +23,24 @@ def convert_path_to_unix_path(path: str) -> str:
         the path in unix format
     """
     return PureWindowsPath(path).as_posix()
+
+
+def check_path_valid_type(path: str) -> bool:
+    """
+    Checks a string to see if is a valid type to be a path, returning false if otherwise
+    Parameters
+    ----------
+    path: str
+        the path to be checked
+
+    Returns
+    -------
+    bool
+        if the string is a valid path type
+    """
+    try:
+        os.path.exists(path)
+        return True
+    except Exception as e:
+        LOG.debug("Exception when trying to use input {} as Path: {}".format(path, e))
+        return False
```

Comment:
What kind of exceptions are we expecting to catch here with `Exception`? Can we scope this down to `TypeError` if its a typing mismatch?